python_bindings/correctness/ext/CMakeLists.txt

```
@@ -18,4 +18,18 @@ foreach (GEN IN ITEMS addconstant bit user_context)
     Python3_add_library(py_${GEN} MODULE ${${GEN}_py_cpp})
     target_link_libraries(py_${GEN} PRIVATE ext_${GEN})
     set_target_properties(py_${GEN} PROPERTIES OUTPUT_NAME ${GEN}) # Python3_add_library adds target info to name.
+
+    # Fake up a linker script that will export *just* the PyInit entry
+    # point we want. (If we don't do this we can have interesting failures
+    # when loading multiple of these Python extensions in the same space.)
+    #
+    # TODO: How to do this for Windows as well?
+    if (APPLE)
+        configure_file(ext.ldscript.apple.in "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript")
+        target_link_options(py_${GEN} PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript")
+    elseif (UNIX)
+        configure_file(ext.ldscript.linux.in "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript")
+        target_link_options(py_${GEN} PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript")
+    endif()
```

We have a helper for this... why not use it?

```suggestion
    include(TargetExportScript)
    configure_file(ext.ldscript.apple.in "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript.apple")
    configure_file(ext.ldscript.linux.in "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript")
    target_export_script(
        py_${GEN}
        APPLE_LD "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript.apple"
        GNU_LD "${CMAKE_CURRENT_BINARY_DIR}/${GEN}.ldscript"
    )
```

---

...because I didn't know about it? :-)